seasmon_xr/accessors.py

```diff
@@ -78,6 +113,60 @@ def _gen_labels(x, lbl, c):
         return f"{x.year}{x.month:02}" + f"{lbl}{int(x.day//c+1)}"
 
 
+class Period(AccessorTimeBase):
+    # pylint: disable=no-member,undefined-variable
+    """
+    Baseclass to extend time dimension with period functionality.
+
+    Adds functionality for working with periods, such as dekads and pentads
+    """
+
+    @property
+    def month_idx(self):
+        return (
+            self._obj.time.to_series()
+            .apply(lambda x: (x.day // self._cfactor) + 1)
+            .values.astype("int")
+        )
+
+    @property
+    def year_idx(self):
+        return (
+            self._obj.time.to_series()
+            .apply(lambda x: ((x.month - 1) * self._nperiods))
+            .values
+            + self.month_idx
+        ).astype("int")
+
+    @property
+    def label(self):
+        return (
+            self.year.astype("str")
+            .str.cat(self.month.astype("str").str.zfill(2))
+            .str.cat(self.month_idx.astype("str"), sep=self._label)
+        ).values
+
+
+@xarray.register_dataset_accessor("dekad")
+@xarray.register_dataarray_accessor("dekad")
+class Dekad(Period):
+    """Accessor class for dekad period."""
+
+    _cfactor = 10.5
+    _nperiods = 3
+    _label = "d"
```

Comment:
set these as "hidden" class variables ... not sure if that's "appropriate"

Reply:
other option is to pass them in to base class constructor at construction time, but this works, only danger is that they don't stay constant....

Reply:
so you're OK with keeping these things as class variables?